topology/types.go

```diff
@@ -255,10 +255,10 @@ type DynamicOptions interface {
 	ServiceID() services.ServiceID
 
 	// SetServiceOptions sets the options for service discovery.
-	SetServiceOptions(opts services.Options) DynamicOptions
+	SetServiceOptions(opts services.OverrideOptions) DynamicOptions
 
 	// ServicesOptions returns the options for service discovery.
-	ServicesOptions() services.Options
+	ServicesOptions() services.OverrideOptions
```

Comment:
The name feels weird. Why `OverrideOptions` instead of `ServicesOptions`?

Reply:
Because we used to have etcd.Options, now that all the code inside etcd/ got moved to services/, I have to rename that to services.Options because that contains all the options to create the SD client. the original services.Options is just an option that allows users to override namespace and stuff, thus the rename, also services.ServicesOptions stutters.

Reply:
Hm should we rename the field in this interface then (`OverrideOptions()` instead of `ServicesOptions()`)?

Reply:
sounds good